### PR TITLE
{system}[system/system] OpenSSL v1.1, OpenSSL v1.0

### LINKED ISSUE #11895: Use wrappers for OpenSSL starting 2021a
As discussed on biweekly meetings, i figured it would be good to make a ticket for it:

Instead of relying on OS, which is getting more and more painful on older OS'es, we should just build OpenSSL (and friends) in EB and keep updating them. This of course requires wrappers, like we do for Java. So, I would expect there to something like a `OpenSSL-1.1.1x-GCCcore-11.a.b.eb` + `OpenSSL-1.1-GCCcore-11.a.b.eb`-wrapper starting with with the 2021 toolchains.

For those wishing to keep osdependency they can filter OpenSSL. We should also make sure to enhance the PR-testsuite so that we don't miss switching over as we bump new easyconfigs to 2021a.

(We might also want to do so for stuff like `gnutls.`, but I suspect osdep works well there still.)

### PR DESCRIPTION
**Pull request by lexming**
(created using `eb --new-pr`)

Builds of OpenSSL at the system level + wrappers.

Depends on ~~https://github.com/easybuilders/easybuild-easyblocks/pull/2429~~

fixes #11895

### CODE REVIEW

**Comment by lexming** [comment]
Test report by @lexming
**FAILED**
Build succeeded for 3 out of 4 (4 easyconfigs in total)
node374.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/455bbef444ee55903de94477574e8458 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
node379.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/f9b6ac5d91c6721d3f49dbee80001655 for a full test report.

**Comment by Micket** [comment]
What about making just an OpenSSL/1.1 that just adds LIBRARY_PATH, CFLAGS, LD_LIBRARY_PATH against the system level libssl.so.1.1?

**Comment by lexming** [comment]
@Micket That can potentially break other EB installations, as loading such module would prepend those generic paths to everything else. We could instead symlink the specific libs and headers. We could even symlink those from the real OpenSSL easyconfig and fool rpaths with it.

**Comment by lexming** [comment]
@Micket I took your suggestion and made an experimental easyblock. I put all the information in https://github.com/easybuilders/easybuild-easyconfigs/issues/11895#issuecomment-841301807

**Comment by lexming** [comment]
Test report by @lexming
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node355.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/ca5858662cafba410344faf3e8436c8c for a full test report.

File: easybuild/easyconfigs/o/OpenSSL/OpenSSL-1.1.eb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+description = """The OpenSSL Project is a collaborative effort to develop a robust, commercial-grade, full-featured,
+ and Open Source toolchain implementing the Secure Sockets Layer (SSL v2/v3) and Transport Layer Security (TLS v1) 
+ protocols as well as a full-strength general purpose cryptography library. """
+
+toolchain = SYSTEM
+toolchainopts = {'pic': True}
+
+osdependencies = [
+    'perl',
+    ('zlib-devel', 'zlib1g-dev'),
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenSSL/OpenSSL-1.1.eb:16]: Are you sure this is required?
Installation works fine of Ubuntu 18.04 without having devel package for `zlib` (or even `zlib` itself) installed.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
ip-172-31-40-8 - Linux ubuntu 18.04, x86_64, Intel(R) Xeon(R) CPU E5-2676 v3 @ 2.40GHz, Python 3.6.9
See https://gist.github.com/39881f3c3f61c93c47740ef0157cad26 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
generoso - Linux UNKNOWN UNKNOWN, x86_64, Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz (haswell), Python 3.8.9
See https://gist.github.com/b0ec992e64572132d32fe4b556ebb10e for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso
EB_ARGS="--include-easyblocks-from-pr 2429"

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
select-pika-c6gd-2xlarge-0001 - Linux centos linux 8.3.2011, AArch64, ARM UNKNOWN (graviton2), Python 3.6.8
See https://gist.github.com/d5cb719ae4bbe02d7cc6c631eb4c9a5c for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
easybuild2.novalocal - Linux centos linux 8.3.2011, POWER, IBM pSeries (emulated by qemu) (power9le), Python 3.6.8
See https://gist.github.com/1da7ae77bf8d8d9ab0ecfacb73105715 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node2644.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/a3595765268647c4333fc7f0436d9726 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3507.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/014adbadd2a91151fb45d421d0ad5a15 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
vpna148.ugent.be - Darwin UNKNOWN UNKNOWN, x86_64, Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz (skylake), Python 3.9.5
See https://gist.github.com/bb62eb966a070ba7eaf6f49795991bb0 for a full test report.

File: easybuild/easyconfigs/o/OpenSSL/OpenSSL-1.1.eb (hunk 1 above)
Comment (lexming) [reply easybuild/easyconfigs/o/OpenSSL/OpenSSL-1.1.eb]: @boegel you are right, the headers are not needed. Fixed in https://github.com/easybuilders/easybuild-easyconfigs/pull/12858/commits/f3385c706220f3fe35734570500a638e3a3623a5

**Comment by Micket** [comment]
Test report by @Micket
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
alvis-c1 - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/769d75de8f16543041ee08b2e0358b87 for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
alvis-c1 - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/bd02020543ecc3ecb77a2b3cfab13f3a for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node378.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/cd81da84f563765b2045e74a97e23899 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
login3.phoenix.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/f8380af33ccf139307b4be6ce8a61a0f for a full test report.

**Comment by lexming** [comment]
* https://github.com/easybuilders/easybuild-easyconfigs/pull/12858#issuecomment-842355634 without `openssl11`
* https://github.com/easybuilders/easybuild-easyconfigs/pull/12858#issuecomment-842356756 with `openssl11`

**Comment by Micket** [comment]
Test report by @Micket
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
alvis-c1 - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/9ae4c02a063dac46e5c523034802bfdd for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node2618.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/f0b2f5160803056fa972248a3b4bb6b6 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3506.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/1a05cfb137937c1d2b1c2d1bce5156e2 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
select-pika-c6gd-2xlarge-0001 - Linux centos linux 8.3.2011, AArch64, ARM UNKNOWN (graviton2), Python 3.6.8
See https://gist.github.com/65726b60cddbb0b2c2cb05ba4b999af3 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
ip-172-31-4-62 - Linux Ubuntu 20.04, AArch64, ARM UNKNOWN, Python 3.8.5
See https://gist.github.com/aba1f7715a054dc313bef108c32f1945 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
ip-172-31-35-128 - Linux ubuntu 18.04, x86_64, Intel(R) Xeon(R) CPU E5-2676 v3 @ 2.40GHz, Python 2.7.17
See https://gist.github.com/5ddcdd41177f5ab876fde8f6d3307fe6 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
easybuild2.novalocal - Linux centos linux 8.3.2011, POWER, IBM pSeries (emulated by qemu) (power9le), Python 3.6.8
See https://gist.github.com/0d5e23aceeb2711b60eb53fc0f0342e5 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
ip-172-31-4-62 - Linux Ubuntu 20.04, AArch64, ARM UNKNOWN, Python 3.8.5
See https://gist.github.com/eed610618a779332ed39c8bae49f2c67 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
ip-172-31-35-128 - Linux ubuntu 18.04, x86_64, Intel(R) Xeon(R) CPU E5-2676 v3 @ 2.40GHz, Python 2.7.17
See https://gist.github.com/4def631b56ef65acb0306ff2b557481e for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
vpnb110.ugent.be - Darwin UNKNOWN UNKNOWN, x86_64, Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz (skylake), Python 3.9.5
See https://gist.github.com/b3a1f5965e9e9a602e0c88c6e83f4858 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by lexming** [comment]
Test report by @lexming
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node378.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/88a51c5873caf8a0b7361a02625d2d9b for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
login3.phoenix.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/18ef371c323e252f6d5afa61f5748ef4 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
login3.phoenix.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/ba101526cb13cf0c06ed4a21248a10ea for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
login3.phoenix.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/71bc298831140056a5ac9ba3f1a08f27 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso
EB_ARGS="--include-easyblocks-from-pr 2429"

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node2618.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/3fd37e6f67ded393bb8f3ff333ee00de for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
select-pika-c6gd-2xlarge-0001 - Linux centos linux 8.3.2011, AArch64, ARM UNKNOWN (graviton2), Python 3.6.8
See https://gist.github.com/2c74a52857a979bef6f35f462efa957f for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3552.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/1d735ab45e55a3e73cba4338b31bd6e5 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
easybuild2.novalocal - Linux centos linux 8.3.2011, POWER, IBM pSeries (emulated by qemu) (power9le), Python 3.6.8
See https://gist.github.com/56f3c047daa2348413846e0547bbbf69 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
ip-172-31-35-128 - Linux ubuntu 18.04, x86_64, Intel(R) Xeon(R) CPU E5-2676 v3 @ 2.40GHz, Python 2.7.17
See https://gist.github.com/ca71143a5c204cb59f0dfc310cfa873c for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2429
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
ip-172-31-4-62 - Linux Ubuntu 20.04, AArch64, ARM UNKNOWN, Python 3.8.5
See https://gist.github.com/77c50c92f02c7fdc828db97d857d99f6 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
Kenneths-MacBook-Pro.local - Darwin UNKNOWN UNKNOWN, x86_64, Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz (skylake), Python 3.9.5
See https://gist.github.com/520b5be0ba3b724bda58b43805b23daa for a full test report.

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Going in, thanks @lexming!

**Pull request merged**